### PR TITLE
Support dynamic volume type quota-sets

### LINKED ISSUE #18: Support dynamic volume type quota-sets
The quota-set api is not fully implemented (https://github.com/openstack4j/openstack4j/blob/master/core/src/main/java/org/openstack4j/openstack/storage/block/domain/CinderBlockQuotaSet.java)
I need to limit the gigabytes per volume type.

api reference:
https://docs.openstack.org/api-ref/block-storage/v3/index.html?expanded=freeze-a-cinder-backend-host-detail,thaw-a-cinder-backend-host-detail,update-quotas-for-a-project-detail#quota-sets-extension-os-quota-sets

### PR DESCRIPTION
**Pull request by manuelmazzuola**
fix #18

### CODE REVIEW

**Comment by olivergondza** [comment]
Thanks for your contribution! It seems the test is failing on travis.

**Comment by manuelmazzuola** [comment]
@olivergondza I've pushed a new commit but travis has not been triggered

**Comment by olivergondza** [comment]
Strange, manual PR rebuild retried from the original commit and not the latest one. Reopening worked, though...

File: core-test/src/main/java/org/openstack4j/api/storage/VolumeTypeTests.java [hunk 1]
```diff
@@ -8,7 +8,9 @@
 import org.openstack4j.api.AbstractTest;
 import org.openstack4j.api.Builders;
 import org.openstack4j.model.common.ActionResponse;
+import org.openstack4j.model.storage.block.BlockQuotaSet;
 import org.openstack4j.model.storage.block.VolumeType;
+import org.openstack4j.openstack.storage.block.domain.CinderBlockQuotaSet;
```
Comment (olivergondza) [inline core-test/src/main/java/org/openstack4j/api/storage/VolumeTypeTests.java]: These are unused after the test was moved.

File: core/src/main/java/org/openstack4j/model/storage/block/builder/BlockQuotaSetBuilder.java [hunk 2]
```diff
@@ -32,4 +32,6 @@
      * @return space consumed in the Block Storage.
      */
     BlockQuotaSetBuilder gigabytes(int gigabytes);
+
+    BlockQuotaSetBuilder volumeTypeQuota(String key, int quota);
```
Comment (olivergondza) [inline core/src/main/java/org/openstack4j/model/storage/block/builder/BlockQuotaSetBuilder.java]: The builder API should accept whole map and not add entries one by one. It is more flexible as that support clearing and removal.

**Comment by olivergondza** [comment]
I gave this a try. JSON response:
```json
{
  "quota_set": {
    "snapshots_shared": -1,
    "per_volume_gigabytes": -1,
    "snapshots_tripleo": -1,
    "volumes_shared": -1,
    "gigabytes": 1200,
    "backup_gigabytes": 100,
    "gigabytes_shared": -1,
    "snapshots": 200,
    "id": "deadbeef",
    "gigabytes_tripleo": -1,
    "snapshots_ceph": -1,
    "gigabytes_ceph": -1,
    "groups": 10,
    "volumes_ceph": -1,
    "backups": 5,
    "volumes_tripleo": -1,
    "volumes": 100
  }
}
```

`getVolumeTypesQuotas()`:
```
backup_gigabytes:100
backups:5
gigabytes_ceph:-1
gigabytes_shared:-1
gigabytes_tripleo:-1
groups:10
per_volume_gigabytes:-1
snapshots_ceph:-1
snapshots_shared:-1
snapshots_tripleo:-1
volumes_ceph:-1
volumes_shared:-1
volumes_tripleo:-1
```

We take all the keys not mapped to fields and serve it from the `getVolumeTypesQuotas()` effectively expecting those are the Volume Types quota. Putting aside this is dangerous as the API can add more keys in the future, `per_volume_gigabytes`, `groups` and `backups` does not seem to belong there.

I find the REST API quite ugly around the dynamic keys, but how about checking the documented prefixes in the `setVolumeTypesQuotas` and only setting those we know are Volume Types quota?

**Comment by manuelmazzuola** [comment]
> I find the REST API quite ugly around the dynamic keys, but how about checking the documented prefixes in the setVolumeTypesQuotas and only setting those we know are Volume Types quota?

you are right, I had started doing that, I have it into a separate branch, but I didn't like it 'cause I ended up to write too much logic inside the dto

I'll resume the code, refactor and push it

**Comment by olivergondza** [comment]
Awesome, thanks for your contribution!

**Pull request merged**